### DISCUSSION TITLE
We should point people to how to write an analyzer.

### DISCUSSION THREAD

**Discussion by AartBluestoke** [dotnet/csharplang#3928] [General] [answered]
Many of the discussion or issues raised in this repo are answered with "just write an analyzer". If this is an intended direction (as is indicated by the growth of FxCop of the years) we should be assisting these people to get the language information they need, rather than telling them to RTFM, without even pointing them to the manual.

I think it would be helpful to have on the front page "Discussions" section (https://github.com/dotnet/csharplang/blob/master/README.md) a statement like ;
- Language extensions for the purpose of preventing specific syntax from occurring can probably be achieved with an analyzer and are likely to be rejected by the language design committee to prevent increased language complexity.
  -- Microsoft outlines how to do so at https://devblogs.microsoft.com/dotnet/how-to-write-a-roslyn-analyzer/
  -- some sample Analyzers can be found at https://github.com/dotnet/roslyn/blob/master/docs/analyzers/Analyzer%20Samples.md
  -- Microsoft's recommended FxCop analyzers can be found at https://github.com/dotnet/roslyn-analyzers

**Comment by svick** [comment]
I think it would be better to point to [the official documentation](https://docs.microsoft.com/en-us/visualstudio/extensibility/getting-started-with-roslyn-analyzers).

**Comment by 333fred** [comment, marked as answer]
I like this suggestion. @AartBluestoke do you want to contribute a pr to modify the Readme? If not I can do it myself, just thought I'd offer since it's your idea.

**Reply by AartBluestoke** [reply]
pr raised.

**Comment by scottdorman** [comment]
> I think it would be better to point to [the official documentation](https://docs.microsoft.com/en-us/visualstudio/extensibility/getting-started-with-roslyn-analyzers).

Would it make sense to *include* the official documentation, using the `[!INCLUDE [<title>](<filepath>)]` [syntax](https://docs.microsoft.com/en-us/contribute/markdown-reference#included-markdown-files) rather than just pointing a link to it? The documentation includes almost all of the links mentioned, and for those that aren't, they could either be added to the official documentation or added to the readme right after the include.

**Reply by 333fred** [reply]
No, I don't think we should do that. Just linking to it is more than sufficient for the README of this repo.
